src/stk/image/volume.cpp

```diff
@@ -316,5 +316,22 @@ void Volume::release()
     _origin = { 0, 0, 0 };
     _spacing = { 1, 1, 1 };
 }
+std::vector<std::string> Volume::get_metadata_keys(void) const
+{
+    std::vector<std::string> keys;
+    keys.reserve(_metadata.size());
+    for (auto const& [k, _] : _metadata) {
+        keys.push_back(k);
```

Comment:
Darn, in 2018 C++ still does not have a damn method to get the key set from a map...